include/snowflake/ClientConfigParser.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2024 Snowflake Computing
+ */
+
+#ifndef PROJECT_CLIENTCONFIG_HPP
```

Comment:
Better to follow the existing naming style. like `SNOWFAKECIENT_CONFIGPARSER_HTTP` ?